core/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceAggregatorBuilder.java

```diff
@@ -82,6 +82,93 @@ protected ValuesSourceAggregatorBuilder(String name, Type type, ValuesSourceType
     }
 
     /**
+     * Read from a stream.
+     */
+    protected ValuesSourceAggregatorBuilder(StreamInput in, Type type, ValuesSourceType valuesSourceType, ValueType targetValueType)
```

Comment:
I know sometimes `valuesSourceType` and `targetValueType` have to be serialized. I think we can handle that in the PR for that builder. I'm confident we can serialize it if we need it though.